pkgs/tools/typesetting/tex/texlive/default.nix

```
@@ -478,12 +478,14 @@ let
     texliveBinaries = bin;
   };
 
-  tl = lib.mapAttrs (pname: { revision, extraRevision ? "", ... }@args:
+  texlivePackages = lib.makeScope newScope (self: lib.mapAttrs (pname: { revision, extraRevision ? "", ... }@args:
```

To understand: why do we need `newScope` here (as opposed to `(x: x)`)? The only difference seems to be the attribute `callPackage`, which we do not use, after all. Everybody else seems to be using `makeScope newScope`.

---

if we don't need/want callPackage, we could use 
```suggestion
  texlivePackages = lib.makeExtensible (self: lib.mapAttrs (pname: { revision, extraRevision ? "", ... }@args:
```
In this case, the "extender"-attribute is called `extend` instead of `overrideScope'`.

---

> "broken" callPackage and newScope attributes.

My problem is probably that I don't quite understand what `newScope` is supposed to do. Is it to make it so the new `callPackage` knows all the attributes from the previous scope?

> Edit: there also is `lib.makeExtensibleWithCustomName`.

Excellent. We want camel case attributes whenever possible to avoid a repeat of the `combine` clash, and `makeScope` injects `packages`, low risk name for sure, but still a risk. If we don't have a use for `newScope` and `callPackage`, it seems like an obvious choice.